test/jest/acceptance/iac/file-output.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { readFileSync, unlinkSync } from 'fs';
+import * as path from 'path';
+import { v4 as uuidv4 } from 'uuid';
+import { startMockServer } from './helpers';
+
+jest.setTimeout(50000);
+
+describe('iac test --json-file-output', () => {
+  let run: (
+    cmd: string,
+  ) => Promise<{ stdout: string; stderr: string; exitCode: number }>;
+  let teardown: () => void;
+
+  beforeAll(async () => {
+    const result = await startMockServer();
+    run = result.run;
+    teardown = result.teardown;
+  });
+
+  afterAll(async () => teardown());
+
+  it('can save JSON output to file while sending human readable output to stdout', async () => {
+    const jsonOutputFilename = path.join(__dirname, `${uuidv4()}.json`);
+    const { stdout } = await run(
+      `snyk iac test ./iac/terraform/sg_open_ssh.tf --json-file-output=${jsonOutputFilename}`,
+    );
+    expect(stdout).toMatch('Organization:');
+
+    const outputFileContents = readFileSync(jsonOutputFilename, 'utf-8');
+    unlinkSync(jsonOutputFilename);
+    const jsonObj = JSON.parse(outputFileContents);
```

Comment:
Can we type the results here? Or at least the parts we're looking for so that we don't have all the casts in the test.
```suggestion
    const jsonObj: ??? = JSON.parse(outputFileContents);
```

Reply:
I copied these tests from `test/jest/acceptance/cli-json-file-output.spec.ts` but I was able to just remove the casts altogether, so we don't have to create a type for the tests

Reply:
I think this might be another todo for later. I think removing the casts for now is fine, but having a type definition for the JSON output would still be useful in maintaining the tests. I think `MappedIacTestResponse` or `IacTestResponse` have most of the fields that are used.

https://github.com/snyk/snyk/blob/2102fbc6c2669b612b98602cc0f584058dec8a17/src/lib/snyk-test/iac-test-result.ts#L79-L82

Reply:
Actually turns out typing exposed that I wasn't doing the right check, so I'll include that in this PR!